Drivers/Comms/Src/Xbee.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Xbee.cpp
+ *
+ *  Created on: Aug 20, 2022
+ *      Author: Aaditya
+ */
+
+#include "Xbee.hpp"
+#include "main.h"
+#include "usart.h"
+
+#define UART_TIMEOUT 100
+
+#define DRONE_BASIC_PAYLOAD_TYPE 1
+
+void changeEndian(uint8_t* val, int size) {
+	uint8_t temp;
+	for (int x = 0; x < size/2; x++) {
+		temp = val[x];
+		val[x] = val[size - 1 - x];
+		val[size - 1 - x] = temp;
+	}
+}
+
+void TransmitMessage(DroneBasicPayload* payload) {
+	Packet packet = {};
+	TransmitFrameHeader tf = {};
+
+	packet.time = HAL_GetTick();
+	packet.payloadType = DRONE_BASIC_PAYLOAD_TYPE;
+
+	uint8_t checksum = 0xFF;
+
+	uint8_t rawPacket[sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) + sizeof(checksum)];
+
+
+	uint8_t* tfArr = (uint8_t*)&tf;
+	uint8_t* packetArr = (uint8_t*)&packet;
+	uint8_t* payloadArr = (uint8_t*)payload;
+
+	tf.messageLength = sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) - 3;
+
+	tf.shortAddress = (tf.shortAddress & 0xFF00) >> 8 | (tf.shortAddress & 0x00FF) << 8;
+
+	changeEndian((uint8_t*)&tf.destAddress, sizeof(tf.destAddress));
```

Comment:
Could you give a TLDR of why the endianness has to be changed here? And maybe add some comments on why as well?

Reply:
TLDR: stm32 uses little endian, but the transmit frame (defined by the xbees protocol) uses big endian. So the data being sent can be in little endian (and is) but the transmit frame header must be in big endian, or the xbee wont be able to read the values. I will leave this as a comment.